node/main.go

```diff
@@ -0,0 +1,213 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/multiformats/go-multiaddr"
+
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/chain/types/ethtypes"
+	"github.com/filecoin-project/lotus/itests/kit"
+	"github.com/filecoin-project/lotus/node/config"
+)
+
+type KitNode struct {
+	FullNode *kit.TestFullNode
+	Miner    *kit.TestMiner
+	Ready    bool
+}
+
+var node KitNode
+var ctx context.Context
+
+func initKit() {
+	blockTime := 1 * time.Second
+	full, miner, ens := kit.EnsembleMinimal(
+		&testing.T{},
+		kit.MockProofs(),
+		kit.ThroughRPC(),
+		kit.WithCfgOpt(func(cfg *config.FullNode) error {
+			cfg.ActorEvent.EnableRealTimeFilterAPI = true
+			return nil
+		}),
+	)
+	ens.InterconnectAll().BeginMining(blockTime)
+	kit.QuietMiningLogs()
+
+	node.FullNode = full
+	node.Miner = miner
+	node.Ready = true
+}
+
+func killKit(ctx context.Context) {
+	if node.Ready {
+		// node.FullNode.Stop(ctx)
+		node.FullNode = nil
+		// node.Miner.Stop(ctx)
+		node.Miner = nil
+		node.Ready = false
+	}
+}
+
+func MultiAddrToHttp(m multiaddr.Multiaddr) (string, error) {
+	sp := strings.Split(m.String(), "/")
+	if len(sp) != 5 {
+		return "", fmt.Errorf("incorrect multiaddr %s", m.String())
+	}
+	host, port := sp[2], sp[4]
+	return fmt.Sprintf("http://%s:%s", host, port), nil
+}
+
+func handleReady(w http.ResponseWriter, req *http.Request) {
+	resp := map[string]interface{}{"ready": node.Ready}
+	output, err := json.Marshal(resp)
+	if err != nil {
+		http.Error(w, err.Error(), 500)
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(output)
+}
+
+func handleRestart(w http.ResponseWriter, req *http.Request) {
+	killKit(ctx)
+	initKit()
```

Comment:
Does this actually work?

Reply:
It worked when I tested it. Did you see anything wrong?